packages/eas-cli/src/commands/branch/create.ts

```diff
@@ -81,7 +81,9 @@ export default class BranchCreate extends EasCommand {
     const projectDir = await findProjectRootAsync();
     const exp = getExpoConfig(projectDir);
     const fullName = await getProjectFullNameAsync(exp);
-    const projectId = await getProjectIdAsync(exp);
+
+    // TODO(wschurman): this non-interactive flag should be standardized
```

Comment:
> Note that this exposed some inconsistencies between how we do non-interactive across commands, so I may unify those before or after landing this, I haven't decided.

Our non-interactivity work is pretty consistent across all the commands, the way it was intended to work is:
- if there is no json/non-interactive flag, we do not want to support enforcing interactivity e.g. `eas credentials` is always interactive so flag there would be redundant, but on the other hand, there are some commands that never needed to prompt for anything so --non-interactive flag would be pointless there too.
- if there is only json flag command it should mean that command is always non-interactive
- it there is only non-interactive flag, it's self-explanatory
- if there are both json and nonInteractive flags, then json flag should require non-interactive to also be set.

In this PR you are making all the commands interactive, and breaking that convention in the process. I could see different approaches how we could resolve it:
- add nonInteractive flag to all commands
- for all commands that do not have nonInteractive flag assume, nonInteractive: true just for the getProjectIdAsync call. With this solution, you need to be careful so the error message does not mention non-interactive flag.
- change convention, I'm open to suggestions
- do not enforce projected on all commands, you could just do that for build/submit/update/update:configure/build:configure.

I'm personally in favor of the second approach

Reply:
Here is one potential sledgehammer approach: https://github.com/expo/eas-cli/pull/1360

I'm also open to the second approach, though I do like the concept of standardization across all our commands, and being able to run all in a non-interactive manner and also optionally output json seems like a good thing.

Reply:
This is a unique case of making all commands interactive in that it theoretically only needs to be run once.

I proposed making the flow such that all commands require having run `eas init` at least once in the past and having that command set this up, but @EvanBacon mentioned that this wasn't a great pattern and that we should instead just do it for them and ask when we are about to make a mutation.

Reply:
Make sense. Do you intend to land #1360 before this PR?

If not, I would still prefer if you consider one of the approaches I suggested as a temporary solution.

Reply:
Going to go with approach 2 for now and then figure out a more general solution (don't want to block this).

Reply:
Alright, looking into this more, I think the better interim approach is to add the non-interactive flag to commands that call `getProjectIdAsync`, `ensureLoggedInAsync`, or other utility methods like them that take an optional nonInteractive argument.

Reply:
Ended up with a hybrid where for each command I assessed whether it was meant to be non-interactive, interactive, or configurable interactive based on the flags. I also made some flag setups consistent (where we have json, also have non-interactive). I also fixed some non-interactive stuff by explicitly passing the flag to functions that use it.